framework/DataObjects/XDataSet.py

```diff
@@ -689,11 +689,16 @@ def _checkAlignedIndexes(self,rlz,tol=1e-15):
     for index in self.indexes:
       # if it's aligned so far, check if it still is
       if index in self._alignedIndexes.keys():
-        # if close enough, then keep the aligned values; otherwise, take action
-        if isinstance(rlz[index][0],(float,int)):
-          closeEnough = all(np.isclose(rlz[index],self._alignedIndexes[index],rtol=tol))
+        # first, if lengths don't match, they're not aligned.
+        if len(rlz[index]) != len(self._alignedIndexes[index]):
+          closeEnough = False
```

Comment:
I have a question here. It seems we will not check the duplicate data in the indexes if there length are not the same. If I understand correctly, this will increase the memory and slow down the calculation. Let's talk on Monday.

Reply:
@alfoa I need you comments here.

Reply:
This is correct; if the length of the indexes is not the same, then the old index and the new index _cannot_ be exactly the same, so we have to store them separately.  Right?

Reply:
As we discussed in the meeting, can you add "FIXME" here regarding the memory and speed issue?

Reply:
done